mk/config.mk

```diff
@@ -262,6 +262,14 @@ CFG_GP_SOCKETS ?= y
 # invocation parameters referring to specific secure memories).
 CFG_SECURE_DATA_PATH ?= n
 
+# Enable storage for TAs in secure storage, depends on CFG_REE_FS=y
+# TA binaries are stored encrypted in the REE FS and are protected by
+# metadata in secure storage.
+CFG_SECSTOR_TA ?= y
```

Comment:
Build fails at link time when `make CFG_REE_FS=n CFG_RPMB_FS=y` (this combination is tested by Shippable). I suggest the following to deal with the dependencies:

```diff
diff --git a/core/core.mk b/core/core.mk
index b633425022d2..e2640b3e0b46 100644
--- a/core/core.mk
+++ b/core/core.mk
@@ -20,12 +20,6 @@ PLATFORM_FLAVOR_$(PLATFORM_FLAVOR) := y
 
 $(call cfg-depends-all,CFG_PAGED_USER_TA,CFG_WITH_PAGER CFG_WITH_USER_TA)
 
-$(call cfg-depends-all,CFG_SECSTOR_TA_MGMT_PTA,CFG_WITH_USER_TA)
-
-ifeq ($(CFG_SECSTOR_TA_MGMT_PTA),y)
-$(call force,CFG_SECSTOR_TA,y,required by CFG_SECSTOR_TA_MGMT_PTA)
-endif
-
 
 # Setup compiler for this sub module
 COMPILER_$(sm)         ?= $(COMPILER)
diff --git a/mk/config.mk b/mk/config.mk
index 2c4e38377665..4835f98a2352 100644
--- a/mk/config.mk
+++ b/mk/config.mk
@@ -265,10 +265,12 @@ CFG_SECURE_DATA_PATH ?= n
 # Enable storage for TAs in secure storage, depends on CFG_REE_FS=y
 # TA binaries are stored encrypted in the REE FS and are protected by
 # metadata in secure storage.
-CFG_SECSTOR_TA ?= y
+CFG_SECSTOR_TA ?= $(call cfg-all-enabled,CFG_REE_FS CFG_WITH_USER_TA)
+$(eval $(call cfg-depends-all,CFG_SECSTOR_TA,CFG_REE_FS CFG_WITH_USER_TA))
 
 # Enable the pseudo TA that managages TA storage in secure storage
-CFG_SECSTOR_TA_MGMT_PTA ?= y
+CFG_SECSTOR_TA_MGMT_PTA ?= $(call cfg-all-enabled,CFG_SECSTOR_TA)
+$(eval $(call cfg-depends-all,CFG_SECSTOR_TA_MGMT_PTA,CFG_SECSTOR_TA))
 
 # Define the number of cores per cluster used in calculating core position.
 # The cluster number is shifted by this value and added to the core ID,
```

The first line (`cfg-all-enabled`) enables the feature by default if dependencies are met, while the second line (`cfg-depends-all`) disables it with a warning in case it is already set but dependencies are not met.
[We could as well fail the build instead of warn and disable ; and write a macro to encapsulate the whole pattern. But that is a topic for a separate PR].

Thoughts?

Reply:
Thanks, I'll fix